release-testing/test

```diff
@@ -1,5 +1,39 @@
 #!/usr/bin/env bash
-set -euxo pipefail
+
+# Set up logging
+#
+# The following setup creates log levels; info and debug. It forces all output, by
+# default, to the debug logfile.
+#
+# 1. The info logfile is simply a file that can be copied to the original stdout by
+# tailing it via a background process (see below) i.e. `tail -f info-log.txt &`
+# 2. Redirect stdout, stderr and xtrace to the debug logfile.
+# 3. Any output for information purposes should be written to the info logfile e.g.
+# `command >> info-log.txt`.
+# 4. To write to both info and debug use tee, making sure to append, i.e.
+# `command | tee -a info-log.txt`.
+
+# Reset logfiles
+printf '' > info-log.txt > debug-log.txt > error-log.txt
+# Begin by tailing the info logfile as a background process so that the contents of the
+# info logfile are output to stdout. A trap, further down, ensures this process is
+# cleaned-up when this script exits.
+tail -f info-log.txt &
+
+# Redirect stdout to append to debug logfile
+exec 1>>debug-log.txt
+# Redirect stderr to append to debug logfile and overwrite to error logfile
+# The error logfile maintains the last write to stderr. It can be used to offer useful
+# context to a failure before digging into the debug logs
+exec 2>> >(tee -a debug-log.txt > error-log.txt)
+# Create a file descriptor for capturing xtrace. This allows us to seperate the xtrace logs
+# from stdout and stderr
+exec 3>>debug-log.txt
```

Comment:
The more I'm looking at this the more it seems like this can (and should) be done with standard log objects in Go rather than manually messing with FDs. Between manually selecting output streams in code and these FDs it's really making the code base have a somewhat overly-complex bespoke feel when it doesn't need to. It's not a blocker for this PR but it should be resolved soon.